kimera_vio_ros/src/interfaces/image_interface.cpp

```diff
@@ -0,0 +1,106 @@
+#include "kimera_vio_ros/interfaces/image_interface.hpp"
+#include "kimera_vio_ros/utils/geometry.hpp"
+
+namespace kimera_vio_ros
+{
+namespace interfaces
+{
+
+ImageInterface::ImageInterface(
+  rclcpp::Node::SharedPtr & node)
+: BaseInterface(node)
+{
+}
+
+ImageInterface::~ImageInterface()
+{
+}
+
+void ImageInterface::msgCamInfoToCameraParams(
+  const CameraInfo::ConstSharedPtr & cam_info,
+  VIO::CameraParams * cam_params)
+{
+  CHECK_NOTNULL(cam_params);
+
+  // Get intrinsics from incoming CameraInfo messages:
+  cam_params->camera_id_ = cam_info->header.frame_id;
+  CHECK(!cam_params->camera_id_.empty());
+
+  CHECK(cam_info->distortion_model == "plumb_bob" ||
+    cam_info->distortion_model == "equidistant");
+
+  if (cam_info->distortion_model == "plumb_bob") {
+    // Kimera-VIO terms the plumb bob dist. model the as radtan.
+    cam_params->distortion_model_ = "radtan";
+    // Kimera-VIO can't take a 6th order radial distortion term.
+    CHECK_EQ(cam_info->d.size(), 5);
```

Comment:
@ToniRV it would be nice if Kimera-VIO understood the `plumb_bob` type to be the same as `radtan`:
* https://github.com/ros/common_msgs/blob/ea6c65e8846d9873b8d6e44435c52fb43d6093a3/sensor_msgs/include/sensor_msgs/distortion_models.h#L45
* https://github.com/ros/common_msgs/blob/ea6c65e8846d9873b8d6e44435c52fb43d6093a3/sensor_msgs/msg/CameraInfo.msg#L58

And also if Kimera-VIO library determined for itself whether to use or not the 6th order radial distortion term.

Also would be cool if Kimera-VIO checked if all the distortion terms where zero, so no rectification would be necessary. Not sure if `cv::initUndistortRectifyMap` internally optimizes for that. Lots of times, ROS image topics will already be rectified to avoid each downstream image consuming node from having to duplicate the rectification. 

https://github.com/ros-perception/vision_opencv/blob/0bbebb73bea65eca856e732b919439853a04986b/image_geometry/src/pinhole_camera_model.cpp#L133-L148